### PR TITLE
fix #292475 - Multiple text elements are added on a chord

### PR DESCRIPTION
**Pull request by njvdberg**
The implementation for list and range selection slightly differs because for
range selections all events are sorted by segment while for list selection
we have to keep track of segments ourselves.

Resolves: https://musescore.org/en/node/292475

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [X] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [X] I made sure the code compiles on my machine
- [X] I made sure there are no unnecessary changes in the code
- [X] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [X] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [X] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
You ticked 
 I made sure the commit message title starts with "fix #424242:" if there is a related issue
But didn't obey to it ;-)

File: mscore/palette.cpp [hunk 1]
```diff
@@ -551,6 +551,39 @@ void Palette::applyPaletteElement(Element* element, Qt::KeyboardModifiers modifi
                   spanner->styleChanged();
                   score->cmdAddSpanner(spanner, idx, startSegment, endSegment);
                   }
+            else if (element->isDynamic() || element->isStaffText() || element->isTempoText() || element->isSystemText()) {
+                  using staffSegment = QPair<Segment*, int>;
+                  QSet<Segment*> segmentSet;
+                  QSet<staffSegment> staffSet;
+                  for (Element* e : sel.elements()) {
+                        Segment* seg = nullptr;
+                        if (e->isNote()) {
```
Comment (Jojo-Schmitz) [inline mscore/palette.cpp]: here and in the next `if'  you're using braces es without need

File: mscore/palette.cpp [hunk 2]
```diff
@@ -695,6 +728,36 @@ void Palette::applyPaletteElement(Element* element, Qt::KeyboardModifiers modifi
...
+                  for (Segment* s = startSegment; s && s != endSegment; s = s->next1()) {
+                        QSet<Staff*> set;
+                        for (int track = track1; track < track2; ++track) {
+                              Element* e = s->element(track);
+                              if (e == 0 || !score->selectionFilter().canSelect(e) || !score->selectionFilter().canSelectVoice(track))
+                                    continue;
+                              if (e->isChordRest()) {
+                                    if (!set.contains(e->staff())) {
+                                          applyDrop(score, viewer,e->isChord() ? toChord(e)->upNote() : e, element, modifiers);
+                                          if (element->isTempoText() || element->isSystemText())
```
Comment (Jojo-Schmitz) [inline mscore/palette.cpp]: Here though you're not using braces, which are not strictly needed here, but might be better as there are 2 lines (even if one is just a comment

Comment (Jojo-Schmitz) [reply mscore/palette.cpp]: Maybe you could move the comment to above of after the `if`?

File: mscore/palette.cpp [hunk 3]
```diff
@@ -695,6 +728,36 @@ void Palette::applyPaletteElement(Element* element, Qt::KeyboardModifiers modifi
...
+                                                // Only one Tempo Text per segment.
+                                                break;
+                                          else
+                                                // Only one Dymanic or Staff Text per segment per track.
```
Comment (Jojo-Schmitz) [inline mscore/palette.cpp]: and this to after the `else`?

File: mscore/palette.cpp (hunk 1 above)
Comment (njvdberg) [reply mscore/palette.cpp]: According the coding style:
Exception 2: Use braces also in if-then-else blocks where either the if-code or the else-code covers several lines:
While the "if" and the "else if" are just single line blocks, the final "else" isn't so this exception applies, or am I mistaken?

File: mscore/palette.cpp (hunk 2 above)
Comment (njvdberg) [reply mscore/palette.cpp]: I tried both options but, personally, I found it was clearer having the comments in the blocks.
And for me it wasn't that clear whether comment were counted as line in the context of "Exception 2". Now it is clear :-)

File: mscore/palette.cpp (hunk 3 above)
Comment (njvdberg) [reply mscore/palette.cpp]: Same comment as above.

File: mscore/palette.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply mscore/palette.cpp]: Not sure, it is a minor nitpick anyhow

**Comment by Jojo-Schmitz** [comment]
Well, now you change the PR title, that's no need, that was fine. And that "fix #NNN" is not case sensitive.
What you should change it the **commit** title

File: mscore/palette.cpp (hunk 2 above)
Comment (njvdberg) [reply mscore/palette.cpp]: Added braces to include the comment in the block.

**Comment by njvdberg** [comment]
Change to commit title to include the issue number. Will remember it for the next pull request :-).

**Comment by Jojo-Schmitz** [comment]
Now you'd just need to squash those 3 commits into one (keeping the 1st's commit message) ;-)

**Comment by njvdberg** [comment]
Squashing commit was a new trick I had to learn. But, done!

**Comment by Howard-C** [comment]
> Squashing commit was a new trick I had to learn. But, done!

Seems like you didn't succeed, there have been piling up more and more commits .

**Comment by njvdberg** [comment]
Indeed, it seems I'm making a mess of it :-( . Trying to clean it up.

**Comment by njvdberg** [comment]
Apparently, besides squashing, there was yet another trick I had to learn. Now there is only one commit.

**Comment by Jojo-Schmitz** [comment]
@njvdberg can you please rebase?

**Comment by njvdberg** [comment]
Rebase done.

**Comment by anatoly-os** [comment]
Postpone due to the @dmitrio95's concern raised in #5560

**Comment by njvdberg** [comment]
I understand the remark of @dmitrio95 as stated in #5560 and I'm willing to have a look whether such a solution can be created.
However, I'm not sure what the best approach is, update this PR or close it and create a new one.

**Comment by vpereverzev** [comment]
It looks like PR is hanging
I propose to make a new one, consideringt the latest changes. Be sure, it will be reviewed faster than this one, lol :)

**Comment by njvdberg** [comment]
A similar [PR #5560](https://github.com/musescore/MuseScore/pull/5560) is merged, let's see this is still necessary.

**Pull request closed without merging**